lib/rubocop/cop/lint/empty_when.rb

```diff
@@ -23,16 +23,38 @@ def on_case(node)
           end
         end
 
+        def comment_lines
+          @comment_lines ||= processed_source.comments.map{|comment| comment.loc.line}
+        end
+
         private
 
         def check_when(when_node)
           return unless empty_when_body?(when_node)
-
           add_offense(when_node, when_node.source_range, MSG)
         end
 
         def empty_when_body?(when_node)
-          !when_node.to_a.last
+          !when_node.to_a.last && no_comments?(when_node)
+        end
+
+        def no_comments?(when_node)
+          (comment_lines & line_range_to_next_node(when_node)).empty?
+        end
+
+        def line_range_to_next_node(node)
```

Comment:
Guess we should really think about this a bit. I thought at some point we had added a way to associate nodes with comments, but perhaps this was just an idea or there was something wrong with the implementation. My memory of this is fuzzy. Maybe we had misunderstood some parser feature... I think it was something like this.